internal/controller/cluster/access_control.go

```diff
@@ -194,14 +204,37 @@ func (r *SingleClusterReconciler) reconcileUsers(
 
 	for userName := range desired {
 		userSpec := desired[userName]
+		currUserSpec, found := current[userName]
+
+		var password *string
+
+		if (!found || currUserSpec.AerospikeAuthMode == asdbv1.AerospikeAuthModeInternal) &&
+			userSpec.AerospikeAuthMode == asdbv1.AerospikeAuthModePKIOnly {
+			if asdbv1.FederalImage(image) {
+				// If federal image and admin user, don't set password.
+				if userName != asdbv1.AdminUsername {
+					nop := ""
+					password = &nop
+				}
+			} else {
+				nop := "nopassword"
+				password = &nop
+			}
+		}
 
-		password, err := passwordProvider.Get(userName, &userSpec)
-		if err != nil {
-			return err
+		if userSpec.AerospikeAuthMode == asdbv1.AerospikeAuthModeInternal {
+			nop, err := passwordProvider.Get(userName, &userSpec)
+			if err != nil {
+				return err
+			}
+
+			password = &nop
 		}
 
+		r.Log.Info("user: ", "userName", userName, "password", password)
```

Comment:
1. Add comment for this block
2. Simplify the overall nested block for better readability. Current implementation is not clear
3. Don't need to check the federal image in very iteration. Can be extracted outside loop